File: 12.数值的整数次方.cpp
class Solution {
public:
    double Power(double base, int exponent) {
        return exponent < 0? 1.0 / helper(base, exponent): helper(base, exponent);
    }
    double helper(double base, int exponent){
        if(exponent == 0)
            return 1.0;
        double ret = helper(base, exponent / 2);
        ret = exponent % 2? ret * ret * base: ret * ret;
        return ret;
    }
};

File: java/12.数值的整数次方.java
public class Solution {
    public static double Power(double base, int exponent) {
        boolean flag = exponent < 0;
        if ( (exponent = Math.abs(exponent)) == 0)
            return 1.0;
        double ret = Power(base, exponent / 2);
        ret = exponent % 2 != 0? ret * ret * base: ret * ret;
        return flag? 1.0 / ret: ret;
    }
}
